### ISSUE TITLE
allow go statements invoke inline function

### ISSUE THREAD

**Issue by renkam** [bombsimon/wsl#111]
Currently for code like this 
```
go func() {
		doSomeStuff()
	}()
```
wsl reports `go statements can only invoke functions assigned on line above`

but here there are no function assignement and it is perfectly clear what is invoked

So please allow this king of invocation with `go` statement

**Comment by bombsimon** [comment]
I think you have something defined on the line above your go statement, I cannot reproduce with your code.

```go
package main

func main() {
        go func() {
                fn()
        }()
}

func fn() {}
```

```sh
$ wsl ./main.go && echo $?
0

$ golangci-lint run --no-config --disable-all --enable wsl && echo $?
0
```

However, if I do something like this:

```go
package main

func main() {
	x := 1
	go func() {
		fn()
	}()

	_ = x
}

func fn() {}
```

I can reproduce it:

```sh'
$ wsl ./main.go && echo $?
./main.go:5: go statements can only invoke functions assigned on line above
```

The solution here would be to add a newline between your assignment and the `go` statement:

```go
package main

func main() {
	x := 1

	go func() {
		fn()
	}()

	_ = x
}

func fn() {}
```

```sh
$ wsl ./main.go && echo $?
0
```

**Comment by renkam** [comment]
There was a variable assigned line before. Indeed removing assignment results with no error.
So this should be really message about cuddle the `go` statement with something else than a function assignment.

**Comment by bombsimon** [comment]
I'll close this since the behaviour is expected. The real issue here is the bad error message. This is addressed in #110 and I don't think it's reasonable to change such behaviour without a new major release, that would f.ex. break existing regex ignores used in `golangci-lint`.

Let me know if you want me to re-open this issue to adress something else!
